packages/apps/plugins/plugin-tree/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+
+  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    switch (event.key) {
+      case 'ArrowUp':
+        // TODO(burdon): Maintain caret position.
+        // TODO(burdon): Go to first/last child of group.
+        onNav?.(item, event.shiftKey ? 'home' : 'up');
+        break;
+      case 'ArrowDown':
+        onNav?.(item, event.shiftKey ? 'end' : 'down');
+        break;
+      case 'Tab':
+        event.preventDefault();
+        onIndent?.(event.shiftKey);
+        break;
+      case 'Enter':
+        // TODO(burdon): Split line (shift to create new).
+        onEnter?.(!!item.text?.length && inputRef.current?.selectionStart === 0);
+        break;
+      case 'Backspace':
+        // TODO(burdon): Merge with previous if caret at start.
+        if (!item.text?.length) {
+          event.preventDefault();
+          onDelete?.();
+        }
+        break;
+      case 'Escape':
+        // TODO(burdon): Toggle checkbox?
+        break;
+    }
+  };
+
+  return (
+    <div className='flex items-center px-2 gap-3'>
+      {(checkbox && (
+        <Input.Root>
+          <Input.Checkbox
+            checked={item.done}
+            onCheckedChange={(checked) => {
+              item.done = !!checked;
+            }}
+          />
+        </Input.Root>
+      )) || (
+        <div className='shrink-0'>
+          <Square weight={active ? 'fill' : undefined} className={mx(getSize(2), active && 'text-primary-500')} />
+        </div>
+      )}
+      <Input.Root>
+        <Input.TextInput
+          ref={inputRef}
+          // autoFocus={active}
+          autoComplete='off'
+          spellCheck={spellCheck}
+          placeholder={focused ? placeholder : undefined}
+          classNames='w-full'
+          variant='subdued'
+          value={item.text ?? ''}
+          onFocus={() => {
+            setFocused(true);
+            onFocus?.();
+          }}
+          onBlur={() => setFocused(false)}
+          onKeyDown={handleKeyDown}
+          onChange={({ target: { value } }) => {
+            item.text = value;
+          }}
+        />
+      </Input.Root>
+      {active && (
+        <DropdownMenu.Root>
+          <DropdownMenu.Trigger asChild>
+            <Button variant='ghost'>
+              <DotsThreeVertical />
+            </Button>
+          </DropdownMenu.Trigger>
+          <DropdownMenu.Portal>
+            <DropdownMenu.Content>
+              <DropdownMenu.Viewport>
+                {onDelete && (
+                  <DropdownMenu.Item onClick={onDelete}>
+                    <X className={getSize(5)} />
+                    <p>Delete item</p>
+                  </DropdownMenu.Item>
+                )}
+              </DropdownMenu.Viewport>
+            </DropdownMenu.Content>
+          </DropdownMenu.Portal>
+        </DropdownMenu.Root>
+      )}
+    </div>
+  );
+};
+
+//
+// Branch
+//
+
+type BranchProps = TreeOptions & {
+  root: Item;
+  active?: string;
+  onFocus?: (item: Item) => void;
```

Comment:
`onFocus` is a native DOM handler name, recommend qualifying this e.g. `onItemFocus`

Reply:
In that case I've onItemXXXd all callbacks.